### PR TITLE
chore: Add a test fixture to validate dynamic Shared VPC

### PR DESCRIPTION
**Pull request by morgante**
This should prevent issues from occurring around the dynamically created/managed subnets or shared VPC host projects.

### CODE REVIEW

**Comment by morgante** [comment]
Issues/Todo:
- [ ] Update to 0.5 release of gcloud module
- [ ] Fix destroy order (service project needs to be destroyed first)

File: test/integration/dynamic_shared_vpc/controls/svpc.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+control 'project-factory-shared-vpc' do
+  title "Project Factory shared VPC"
+
+  describe command("gcloud compute shared-vpc get-host-project #{service_project_id} --format='get(name)'") do
+    its('exit_status') { should eq 0 }
+    its('stderr') { should eq '' }
+    its('stdout.strip') { should eq shared_vpc }
+  end
+
+  describe command("gcloud projects get-iam-policy #{shared_vpc} --format=json") do
```
Comment (rjerrems) [inline test/integration/dynamic_shared_vpc/controls/svpc.rb]: Not a blockers, but It should be possible to do this directly in inspec:

https://www.inspec.io/docs/reference/resources/google_project_iam_bindings/

File: test/integration/dynamic_shared_vpc/controls/svpc.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    it "includes the GKE service account in the roles/container.hostServiceAgentUser IAM binding" do
+      expect(bindings).to include(
+        members: including("serviceAccount:service-#{service_project_number}@container-engine-robot.iam.gserviceaccount.com"),
+        role: "roles/container.hostServiceAgentUser",
+      )
+    end
+  end
+
+  describe command("gcloud beta compute networks subnets get-iam-policy #{shared_vpc_subnet_name_01} --region #{shared_vpc_subnet_region_01} --project #{shared_vpc} --format=json") do
```
Comment (rjerrems) [inline test/integration/dynamic_shared_vpc/controls/svpc.rb]: Similar to above https://www.inspec.io/docs/reference/resources/google_compute_subnetwork_iam_policy/

File: test/integration/dynamic_shared_vpc/controls/svpc.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      it "includes the project service account in the roles/compute.networkUser IAM binding" do
+        expect(bindings).to include(
+          members: including("serviceAccount:#{service_account_email}"),
+          role: "roles/compute.networkUser",
+        )
+      end
+    end
+  end
+
+  describe command("gcloud beta compute networks subnets get-iam-policy #{shared_vpc_subnet_name_02} --region #{shared_vpc_subnet_region_02} --project #{shared_vpc} --format=json") do
```
Comment (rjerrems) [inline test/integration/dynamic_shared_vpc/controls/svpc.rb]: Same as above https://www.inspec.io/docs/reference/resources/google_compute_subnetwork_iam_policy/

**Comment by morgante** [comment]
> LGTM, only feedback is that we could use native inspec for some of the tests - we could track that as a separate issue as we have other tests that could be updated or is there a specific reason we are using gcloud commands directly?

At the time, gcloud commands were the only option. TBH, given available resourcing I don't think it's worth spending time/effort to switch over - they're functionally equivalent so no added value.

**Pull request merged**